### PR TITLE
Use JSONIntStr for user endpoint

### PR DESCRIPTION
**Pull request by moltzaum**
#### What does this PR do?

Has a temporary fix for #3304 and is just for the users endpoint. Thinking a better solution exists for all endpoints, but this is to wrap up the current TC-verification process.

#### Which TC components are affected by this PR?

- [ ] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] Other _________

#### What is the best way to verify this PR?
POST a user with "" for the role or tenant id. I've presumed the gid and uid don't need to use JSONIntStr, nor the main id.

#### Check all that apply

- [ ] This PR includes tests
- [ ] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [ ] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

File: traffic_ops/client/user.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	"strconv"
 
 	"github.com/apache/trafficcontrol/lib/go-tc"
+	util "github.com/apache/trafficcontrol/lib/go-util"
```
Comment (rob05c) [inline traffic_ops/client/user.go]: Nitpick: the alias isn't necessary, `go-util` package automatically imports as `util`

Comment (moltzaum) [reply traffic_ops/client/user.go]: That was done automatically when I saved. ¯\_(ツ)_/¯

Comment (moltzaum) [reply traffic_ops/client/user.go]: I think it does that  because the go convention is for the package name to match the folder name.

**Comment by rawlinp** [comment]
Doesn't this break the golang client from 2.2.x? https://github.com/apache/trafficcontrol/blob/2.2.x/lib/go-tc/users.go

If so, Go TO client users should be notified somehow.

**Comment by moltzaum** [comment]
It seems to break whatever might be using the `go-tc` structs on an upgrade to 3.0.x. Anyone referencing the Role and TenantID field in particular will need to cast to an int while using it.

**Comment by rob05c** [comment]
I see two options:
1. leave the code, and add a note to the CHANGELOG.md
2. Change `tc.User` back to `int`s, and anonymously embed a `JSONIntStr` in the TO code, e.g.
```
type TOUser struct {
	ReqInfo *api.APIInfo `json:"-"`
	tc.User

	Role            *util.JSONIntStr `json:"role" db:"role"`
	TenantID        *util.JSONIntStr `json:"tenantId" db:"tenant_id"`
}

⋮ 

if user.Role != nil {
  user.User.Role = util.IntPtr(*user.Role)
}
```

I'd lean toward the latter. It's not that much code on our side, and it provides a cleaner user experience. And besides, we're already doing it in other places, e.g. where we need a `pq.Array`.

**Pull request merged**